tests/oc_template_test.go

```diff
@@ -0,0 +1,328 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/testutils"
+	"kubevirt.io/kubevirt/pkg/virt-controller/services"
+	"kubevirt.io/kubevirt/tests"
+)
+
+// template parameters
+const (
+	vmName     = "testvm"
+	vmCpuCores = "2"
+)
+
+// parameter strings
+const (
+	vmTemplateParams = "-p NAME=" + vmName + " -p CPU_CORES=" + vmCpuCores
+	vmStartPatch     = "-p '{\"spec\":{\"running\":true}}'"
+	vmStopPatch      = "-p '{\"spec\":{\"running\":false}}'"
+)
+
+// oc-related options
+const (
+	ocUser = "system:admin"
+)
+
+var _ = Describe("VM Template", func() {
+	flag.Parse()
+
+	tests.PanicOnError(err)
+
+	// I need to figure out how to generate this JSON string from a struct using tests/types.go
```

Comment:
What exactly is blocking you there? If it is about the serialization process itself, have a look at https://blog.golang.org/json-and-go. That should contain everything needed.